packages/otel/src/vercel-resource-attributes.ts

```diff
@@ -0,0 +1,115 @@
+/**
+ * DataDog resource attributes for @vercel/otel compatibility.
+ *
+ * ## Why this file exists
+ *
+ * @vercel/otel's CompositeSpanProcessor adds DataDog-specific attributes
+ * (`operation.name`, `resource.name`) in onEnd() - but ONLY for sampled spans.
+ *
+ * When using tail sampling, spans start as non-sampled (RECORD decision) and
+ * get promoted to SAMPLED by TailSamplingProcessor. The problem is:
+ *
+ * 1. Span ends → CompositeSpanProcessor.onEnd() runs first
+ * 2. CompositeSpanProcessor checks isSampled() → false → skips resource attributes
+ * 3. TailSamplingProcessor.onEnd() runs → promotes span to SAMPLED
+ * 4. Promoted span goes to BatchSpanProcessor (never back through CompositeSpanProcessor)
+ *
+ * Result: promoted spans are missing DataDog attributes that normally sampled spans have.
+ *
+ * ## Why CompositeSpanProcessor runs first
+ *
+ * registerOTel() from @vercel/otel wraps our spanProcessors inside CompositeSpanProcessor:
+ *
+ * ```
+ * tracerProvider.addSpanProcessor(
+ *   new CompositeSpanProcessor(spanProcessors, ...)  // Our processors nested inside
+ * )
+ * ```
+ *
+ * This creates the processor chain:
+ *   SDK → CompositeSpanProcessor → TailSamplingProcessor → BatchSpanProcessor
+ *
+ * ## Solution
+ *
+ * We replicate @vercel/otel's getResourceAttributes() logic here so that
+ * TailSamplingProcessor can add these attributes when promoting spans.
```

Comment:
do you have to replicate it?

can you 

```
class TailSamplingProcess {
  constrcutor(private vercelProcessor: CompositeSpanProcessor ){}

   promoteToSampled() {
  ... 

   this.vercelPRocessor.something()...
}
}
```

Reply:
new version also doesn't export this: https://github.com/vercel/otel/blob/main/packages/otel/src/processor/composite-span-processor.ts#L150